Pod/Core/XCTestCase+Gherkin.swift

```diff
@@ -20,6 +20,9 @@ This means that anytime I want to access my extra properties I just do `state.{p
 class GherkinState: NSObject, XCTestObservation {
     var test: XCTestCase?
     
+    // Arbitrary user data associated with the current test scenario
+    var scenarioContext: [String: MatchedStringRepresentable] = [:]
```

Comment:
Why `MatchedStringRepresentable` and not just `Any`?

Reply:
It is to make it work together with example values. It may be handy to be able to "interpolate" the value from state set by some previous step the same way as with example value (there is an example of that in the tests). I think having it as step input makes it a bit more explicit when you read the test (like method dependency injection vs accessing property inside the method).
The protocol name makes it a bit confusing though.